partner_vat_required/models/res_users.py

```diff
@@ -0,0 +1,16 @@
+# © 2022 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, api
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    @api.model
+    def create(self, vals):
+        """
+        Prevent required VAT for contact linked with user
+        """
+        users = self.with_context(skip_vat_required=True)
+        return super(ResUsers, users).create(vals)
```

Comment:
For v17 users creation is based on multi creation (https://github.com/odoo/odoo/blob/5c20571c3cddf2aba7de87632e88443629d08417/odoo/addons/base/models/res_users.py#L669). Suggested code, not tested:
```suggestion
    @api.model_create_multi
    def create(self, vals_list):
        """
        Prevent required VAT for contact linked with user
        """
        users = self.with_context(skip_vat_required=True)
        return super(ResUsers, users).create(vals_list)
```